rules/correct-component-name.js

```diff
@@ -0,0 +1,109 @@
+const { dim, bold } = require('colors');
+const { Rule } = require('../api');
+
+module.exports = class extends Rule {
+    getName() {
+        return 'correct-component-name';
+    }
+
+    checkTwigFile(component) {
+        const { twig } = component.files;
+        const {name: twigFileName} = twig;
+        const configDefinition = twig.api.external.definitions.filter(definition => definition.name === 'config')[0];
+        const { type, name, path } = component;
+        let contract;
+        let configNameStringIndex;
+        let configName;
+
+        if (configDefinition) {
+            contract = configDefinition.contract;
+            configNameStringIndex = contract.indexOf('name:');
+
+            if (configNameStringIndex >= 0) {
+                const contractStringFromNameProperty = contract.slice(configNameStringIndex + contract.slice(configNameStringIndex).indexOf('\'') + 1);
+                configName = contractStringFromNameProperty.slice(0, contractStringFromNameProperty.indexOf('\''));
+            }
+        }
+
+        if (name !== twigFileName.slice(0, twigFileName.lastIndexOf('.twig'))) {
+            this.outcome.addError(`There is wrong name of twig file in ${type} ${bold(name)}:\n${dim(path)}`);
+        }
+
+        if (!configDefinition) {
+            this.outcome.addError(`There is no config block in twig file of ${type} ${bold(name)}:\n${dim(path)}`);
```

Comment:
I think this check is not related to component name, I think we need some basic checks, like `correct-twig-structure`, `correct-scss-structure` and `correct-ts-structure`, where we can check if component contains `config` and `data` objects, if main block name is `body`, etc.

Reply:
I created separate rules for checking structure of component twig, scss and typescript files